### ISSUE TITLE
`isolatedDeclarations` false positive TS9013 on arrow functions with default parameters and `as const` objects

### ISSUE THREAD

**Issue by jakeboone02** [microsoft/typescript-go#3207]
This issue was discovered on the https://github.com/react-querybuilder/react-querybuilder repo.

## Steps to reproduce

```ts
// repro.ts
const imported = ',';

export const fn1 = (x: string = imported): string => x;

interface Opts { val?: boolean }
export const fn2 = ({ val = false }: Opts = {}): boolean => val;

export const obj1 = { $and: [{ $expr: true }] } as const;
```

```jsonc
// tsconfig.json
{
  "compilerOptions": {
    "noEmit": true,
    "declaration": true,
    "isolatedDeclarations": true,
    "strict": true,
    "target": "esnext",
    "module": "esnext",
    "moduleResolution": "bundler"
  },
  "include": ["repro.ts"]
}
```

## Behavior with `typescript@6.0.2`

```
$ tsc -p tsconfig.json
// No errors
```

## Behavior with `tsgo` (7.0.0-dev.20260324.1)

```
$ tsgo -p tsconfig.json
repro.ts:3:21 - error TS9013: Expression type can't be inferred with --isolatedDeclarations.
repro.ts:6:21 - error TS9013: Expression type can't be inferred with --isolatedDeclarations.
repro.ts:8:29 - error TS9013: Expression type can't be inferred with --isolatedDeclarations.
repro.ts:8:39 - error TS9013: Expression type can't be inferred with --isolatedDeclarations.

Found 4 errors in the same file, starting at: repro.ts:3
```

## Additional context

These appear to be regressions from #2459 (isolatedDeclarations port, merged 2026-03-20). The same class of false positive was previously reported and fixed in tsc:

- [TypeScript#60192](https://github.com/microsoft/TypeScript/issues/60192) — false positive on arrow functions with typed optional/default parameters
- [TypeScript#60195](https://github.com/microsoft/TypeScript/pull/60195) — the fix

Two distinct patterns are affected:
1. **Arrow functions with default parameters** — `export const fn = (param: Type = expr): ReturnType => ...` is flagged even though the parameter already has a type annotation
2. **`as const` with nested object/array literals** — `export const obj = { key: [{ nested: true }] } as const` is flagged

**Comment by hkleungai** [comment]
> **`as const` with nested object/array literals**

Seems to be a known issue on this one. Ref: https://github.com/microsoft/typescript-go/issues/3192#issuecomment-4105762522

Edit: They may not be identical issue though. But we now know for this situation, tsgo behaves unexpectedly at both type emit and `noEmit` under `isolatedDeclarations` flag.

**Comment by jakebailey** [comment]
The first is is an easy fix. I have not looked at the latter.
